schemas/2015-01-01/deploymentTemplate.json

```diff
@@ -81,6 +83,8 @@
               {
                 "oneOf": [
                   { "$ref": "http://schema.management.azure.com/schemas/2015-06-01/Microsoft.KeyVault.json#/resourceDefinitions/vaults" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2015-06-01/Microsoft.Cdn.json#/resourceDefinitions/profiles" },
```

Comment:
It is not only about RP's resource types hierarchy, it is about if you want to support a "resources" property under a parent resource.
Please take a look at "resourceDefinitions\servers\resources", "resourceDefinitions\databasesChild", "resourceDefinitions\databases" in "2014-04-01-preview/Microsoft.Sql.json"